### PR TITLE
ISPN-5673 Observation attribute is ignored for listeners with indexed DSL filters

### PR DESCRIPTION
**Pull request by anistor**
jira: https://issues.jboss.org/browse/ISPN-5673,
       https://issues.jboss.org/browse/ISPN-5669

### CODE REVIEW

**Comment by anistor** [comment]
CI failures are not related.

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java [hunk 1]
```diff
@@ -245,6 +246,9 @@ public void onFilterResult(Object userContext, Object instance, Object eventType
                event = clone;
             }
             for (DelegatingCacheEntryListenerInvocation<K, V> invocation : invocations) {
+               if (event.isPre() && invocation.getObservation() == Listener.Observation.POST) {
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java]: For completeness do we want the inverse? When event is post and observation is PRE ?

Comment (anistor) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java]: Indeed! I'll fix in a few mins.

Comment (anistor) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java]: Updated.

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java [hunk 2]
```diff
@@ -245,6 +246,15 @@ public void onFilterResult(Object userContext, Object instance, Object eventType
                event = clone;
             }
             for (DelegatingCacheEntryListenerInvocation<K, V> invocation : invocations) {
+               if (event.isPre()) {
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java]: I thought about this more.  The listener always has an observation.  Maybe we can just do?
```
if (invocation.getObservation().shouldInvoke(event.isPre())) {
   invocation.invokeNoChecks(event, false, filterAndConvert);
}
```

We may have to change https://github.com/infinispan/infinispan/pull/3643/files#diff-896a626d3ff4076e967def69ba12d01dR329 to return BOTH as well maybe.

Comment (anistor) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java]: Good advice! Fixed and updated.

**Comment by anistor** [comment]
CI failures are not related.

**Comment by wburns** [comment]
Pulling..

**Comment by wburns** [comment]
Integrated into master, thanks @anistor !

**Pull request closed without merging**